src/goto-checker/goto_symex_fault_localizer.cpp

```diff
@@ -0,0 +1,146 @@
+/*******************************************************************\
+
+Module: Fault Localization for Goto Symex
+
+Author: Peter Schrammel
+
+\*******************************************************************/
+
+/// \file
+/// Fault Localization for Goto Symex
+
+#include "goto_symex_fault_localizer.h"
+
+#include <solvers/prop/prop_conv.h>
+
+goto_symex_fault_localizert::goto_symex_fault_localizert(
+  const optionst &options,
+  ui_message_handlert &ui_message_handler,
+  const symex_target_equationt &equation,
+  prop_convt &solver)
+  : options(options),
+    ui_message_handler(ui_message_handler),
+    equation(equation),
+    solver(solver)
+{
+}
+
+fault_location_infot goto_symex_fault_localizert::
+operator()(const irep_idt &failed_property_id)
+{
+  fault_location_infot fault_location;
+  localization_pointst localization_points;
+  const auto &failed_step =
+    collect_guards(failed_property_id, localization_points, fault_location);
+
+  if(!localization_points.empty())
+  {
+    messaget log(ui_message_handler);
+    log.status() << "Localizing fault" << messaget::eom;
+
+    // pick localization method
+    //  if(options.get_option("localize-faults-method")=="TBD")
```

Comment:
I suppose this is copy&paste?

Reply:
Yes, I am inclined to remove this option altogether as nobody has ever implemented any alternative method in 3 years.